flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -684,6 +689,96 @@ private void setCoLocation() {
 		}
 	}
 
+	private void setManagedMemoryFraction() {
+		// all slot sharing groups in this job
+		final Set<SlotSharingGroup> slotSharingGroups = Collections.newSetFromMap(new IdentityHashMap<>());
+		// maps a job vertex ID to its head operator ID
+		final Map<JobVertexID, Integer> vertexHeadOperators = new HashMap<>();
+		// maps a job vertex ID to IDs of all operators in the vertex
+		final Map<JobVertexID, Set<Integer>> vertexOperators = new HashMap<>();
+
+		for (Entry<Integer, JobVertex> entry : jobVertices.entrySet()) {
+			final int headOperatorId = entry.getKey();
+			final JobVertex jobVertex = entry.getValue();
+
+			final SlotSharingGroup jobVertexSlotSharingGroup = jobVertex.getSlotSharingGroup();
+			if (jobVertexSlotSharingGroup != null) {
+				slotSharingGroups.add(jobVertexSlotSharingGroup);
+			}
+
+			vertexHeadOperators.put(jobVertex.getID(), headOperatorId);
+
+			final Set<Integer> operatorIds = new HashSet<>();
+			operatorIds.add(headOperatorId);
+			operatorIds.addAll(chainedConfigs.getOrDefault(headOperatorId, new HashMap<>()).keySet());
+			vertexOperators.put(jobVertex.getID(), operatorIds);
+		}
+
+		for (SlotSharingGroup slotSharingGroup : slotSharingGroups) {
+			setManagedMemoryFractionForSlotSharingGroup(slotSharingGroup, vertexHeadOperators, vertexOperators);
+		}
+	}
+
+	private void setManagedMemoryFractionForSlotSharingGroup(
+			final SlotSharingGroup slotSharingGroup,
+			final Map<JobVertexID, Integer> vertexHeadOperators,
+			final Map<JobVertexID, Set<Integer>> vertexOperators) {
+
+		int groupOperatorCount = 0;
+		for (JobVertexID jobVertexID : slotSharingGroup.getJobVertexIds()) {
+			final Set<Integer> operatorIds = vertexOperators.get(jobVertexID);
+			groupOperatorCount += operatorIds.size();
+		}
+
+		for (JobVertexID jobVertexID : slotSharingGroup.getJobVertexIds()) {
+			for (int operatorNodeId : vertexOperators.get(jobVertexID)) {
+				final StreamConfig operatorConfig = vertexConfigs.get(operatorNodeId);
+				final ResourceSpec operatorResourceSpec = streamGraph.getStreamNode(operatorNodeId).getMinResources();
+				setManagedMemoryFractionForOperator(
+					operatorResourceSpec,
+					slotSharingGroup.getResourceSpec(),
+					groupOperatorCount,
+					operatorConfig);
+			}
+
+			// need to refresh the chained task configs
```

Comment:
include the reason; because they are serialized

Reply:
done.